apps/shipments/views.py

```diff
@@ -101,24 +101,63 @@ def get_tracking_data(self, request, version, pk):
 
         if tracking_data.exists():
             response = Template('{"data": $geojson}').substitute(geojson=render_point_features(shipment, tracking_data))
-            return HttpResponse(content=response,
-                                content_type='application/vnd.api+json')
+            httpresponse = HttpResponse(content=response, content_type='application/vnd.api+json')
+            return httpresponse
 
         return HttpResponseNotFound()
 
+    @method_decorator(cache_page(60 * 10, cache='page'))  # Cache responses for 10 minutes
+    @action(detail=True, methods=['get'], permission_classes=(IsOwnerOrShared,))
+    def tracking(self, request, version, pk):
+        return self.get_tracking_data(request, version, pk)
```

Comment:
You can move the contents of get_tracking_data here, there's no reason to have it separate now